src/infrastructure/server.ts

```diff
@@ -1,12 +1,12 @@
 'use strict';
 
-import Hapi from '@hapi/hapi';
-import { Server } from '@hapi/hapi';
+import { Config } from '../config/environmentConfig';
+import Hapi, { Server } from '@hapi/hapi';
 
 // Create the hapi server
 const server: Server = Hapi.server({
-  host: process.env.HOST || 'localhost',
-  port: process.env.NODE_ENV !== 'test' ? process.env.PORT || 3000 : 4000,
+  host: process.env.HOST ?? 'localhost',
+  port: Config.env !== 'test' ? Config.port : 4000,
```

Comment:
May i know the reason for running the app on different port on the TEST env?

Reply:
Jest uses the `test` environment while running the test scenarios. Jest test is similar to browser testing, so some test cases may behave weird now or in the future if the local server is running on 3000. An address conflict error will occur if jest use the same port. To avoid this, I have another port for it.